src/libsystemd-network/sd-ipv4acd.c

```diff
@@ -453,6 +455,28 @@ int sd_ipv4acd_set_mac(sd_ipv4acd *acd, const struct ether_addr *addr) {
         return 0;
 }
 
+int sd_ipv4acd_add_other_mac(sd_ipv4acd *acd, const struct ether_addr *addr) {
+        struct ether_addr *a;
+
+        assert_return(acd, -EINVAL);
+        assert_return(addr, -EINVAL);
+        assert_return(!ether_addr_is_null(addr), -EINVAL);
+
+        a = newdup(struct ether_addr, addr, 1);
+        if (!a)
+                return -ENOMEM;
+
+        return set_ensure_consume(&acd->other_mac_addrs, &ether_addr_hash_ops_free, a);
+}
+
+int sd_ipv4acd_remove_other_mac(sd_ipv4acd *acd, const struct ether_addr *addr) {
+        assert_return(acd, -EINVAL);
+        assert_return(addr, -EINVAL);
+
+        free(set_remove(acd->other_mac_addrs, addr));
+        return 0;
+}
```

Comment:
i wonder if this shouldn't be done with a callback instead. I.e. instead of propagating the MAC addresses of the whole system continously into all sd_ipv4acd objects, do it the other way round: allow configuring a callback function that is called whenver a MAC address should be checked if local. And then maintain a single global hashmap inside of networkd where we keep track of all local MAC addreses and the interfaces they belong to.

This could even be useful to detect MAC address collisions when automatically assigning a MAC address or so.

Or in other words:

1. Extend the `Manager` object to gain a new field `links_by_mac_address`. It's keyed by the mac address and points to the interface with that MAC address, as is kept track of the same way as `links_by_name` is. (if we want to support that the same MAC address may be set for multiple local interfaces, then I guess we need to make sure that links with the same one are linked up in a linked list, and the `links_by_mac_address` hashmap points only to the first of them.

2. Add sd_ipv4acd_set_other_mac_callback() that takes an `sd_ipv4acd_other_mac_callback_t` parameter which is a function that returns 0, 1 or error, and gets a `const struct ether_addr` as parameter.

3. Add glue code to networkd to implement sd_ipv4acd_other_mac_callback_t once, and make it check the `links_by_mac_address` hashmap.

I think this would be more robust, less duplication and use less resources.

(Btw, I think it would be nice to rename the `links` field in `Manager` to `links_by_ifindex`, so that we then nicely have `links_by_ifindex`, `links_by_ifname` and `links_by_mac`)